src/Data/Text/Lazy.hs

```diff
@@ -1427,18 +1429,32 @@ groupBy eq (Chunk t ts) = cons x ys : groupBy eq zs
 -- | /O(n)/ Return all initial segments of the given 'Text',
 -- shortest first.
 inits :: Text -> [Text]
-inits = (Empty :) . inits'
+inits = (NE.toList P.$!) . initsNE
+
+-- | /O(n)/ Return all initial segments of the given 'Text',
+-- shortest first.
+--
+-- @since 2.1.2
+initsNE :: Text -> NonEmpty Text
+initsNE = (Empty NE.:|) . inits'
   where inits' Empty        = []
         inits' (Chunk t ts) = L.map (\t' -> Chunk t' Empty) (L.drop 1 (T.inits t))
```

Comment:
```suggestion
        inits' (Chunk t ts) = L.map (\t' -> Chunk t' Empty) (NE.tail (T.initsNE t))
```

`L.drop` does not inline because it is recursive.